frame/vamm/src/tests/swap.rs

```diff
@@ -332,13 +332,40 @@ fn should_not_update_twap_if_current_twap_timestamp_is_more_recent() {
 		assert_ne!(vamm_state_t0.twap_timestamp, vamm_state_t1.twap_timestamp);
 
 		// In the second swap the time didn't pass between operations, so we
-		// can't allow the twap value to be updated again.
+		// can't allow the twap timestamp nor the twap value to be updated
+		// again.
 		assert_ok!(TestPallet::swap(&swap_config));
 		let vamm_state_t2 = TestPallet::get_vamm(0).unwrap();
 		assert_eq!(vamm_state_t1.twap_timestamp, vamm_state_t2.twap_timestamp);
+		assert_eq!(vamm_state_t1.base_asset_twap, vamm_state_t2.base_asset_twap);
```

Comment:
What happens if the price changes rapidly within the same block? Shouldn't we allow the TWAP to change a bit in that case? Otherwise, the TWAP in a block will be determined by the first swap in it, and subsequent swaps would be free to be as large as they please without changing the TWAP.

I have a feeling that's why other protocols have a minimum weight always for the latest price, even if little to no time has passed since the last swap. DM me if you want to discuss

```text
                                                       twap_period
                                        |---------------------------------------|
                                         from_start          since_last
                                        |-----------|---------------------------|
 -------------------------------------------------------------------------------|
                                        ^           ^                           ^
                                      now -         |                          now
                                   twap_period      |
                                               last_twap_ts
```